config/src/config/test_data/validator_full_node.yaml

```diff
@@ -32,3 +32,7 @@ full_node_networks:
       seed_peers:
           "c227da54069989f283712e4016704660":
               - "/ip4/127.0.0.1/tcp/58259/ln-noise-ik/c998dcd54c3daf48e0ad516d94b7be0b0b7a27caa00541f2b2c14b13500df10b/ln-handshake/0"
+
+upstream:
+    networks:
+        - private: vfn
```

Comment:
is the public network defined here the one used to failover to other on-chain discovered network, or the one advertised on-chain? how are we differentiating b/w the two public networks? because one of them has to be defined in the upstream config (the one where you failover to another publicly advertised VFN)

Reply:
the public network defined here `is` the one that's advertised on-chain

Reply:
and this will be the same network used to for this VFN to fallback to other publicly advertised VFN?
then we have a situation where the same network identity is used by this VFN to communicate with downstream nodes and the upstream failover VFN, and I don't think that'll work for upstream config?

Reply:
Ahh... hmmm... this is interesting... and now I see a flaw in how we got here... we actually can end up in a strange split-case where we want 2 public networks... one for incoming clients and one for connecting to other VFNs.

I think the solution is to actually support an optional classifier for the public network that lets us distinguish these use cases... so that the on wire NetworkId is the same, but inside the code it allows for two distinct networking stacks. So effectively, we could imagine Public(bool) where bool defaults to false and true would denote this redundancy.

Reply:
not familiar with networking code, but I assume it's possible to differentiate b/w which peers are incoming and outgoing dials? we would have to map that to true/false and propagate that to application layer (mempool/state sync), maybe in network context? so an incoming public peer would come with NetworkId::Public(true) or something like this

Reply:
that might work as well! we already get NetworkId in the mempool/state sync via the NewPeer event, right? So we'd just be adding another network. AFAIU, right?

Reply:
lol, that's the ugliness of using p2p model for upstream/downstream, time to consider client/server model?

Reply:
distinguish inbound/outbound connections has to work in validator network too i.e. a validator should treat both inbound/outbound connection as upstream while a full node should not

Reply:
at least in our current code, validator network already has its own NetworkId::Validator that's always treated upstream, so we don't have to worry about distinguishing inbound/outbound for that (validator will just ignore that concept)

> time to consider client/server model?
if only we had more time.. T_T

for now.. I guess I can push a PR for adding this true/false variations of NetworkId::Public, set default value as placeholder and pipe that out from NewPeer event to mempool/state sync, and someone on networking side can actually make sure the correct variant of NetworkId::Public gets passed to NewPeer? we'll basically meet each other halfway (again)

or if there is another way to organize these two networking stacks, I am also open to other options, this was just what came to my head immediately